ledger/participant-integration-api/src/main/scala/platform/apiserver/StandaloneApiServer.scala

```diff
@@ -36,45 +36,118 @@ import scalaz.{-\/, \/-}
-    metrics: Metrics,
-    timeServiceBackend: Option[TimeServiceBackend] = None,
-    otherServices: immutable.Seq[BindableService] = immutable.Seq.empty,
-    otherInterceptors: List[ServerInterceptor] = List.empty,
-    engine: Engine,
-    servicesExecutionContext: ExecutionContextExecutor,
-    lfValueTranslationCache: LfValueTranslationCache.Cache,
-    checkOverloaded: TelemetryContext => Option[state.SubmissionResult] =
-      _ => None, // Used for Canton rate-limiting
-)(implicit actorSystem: ActorSystem, materializer: Materializer, loggingContext: LoggingContext)
-    extends ResourceOwner[ApiServer] {
-
+object StandaloneApiServer {
   private val logger = ContextualizedLogger.get(this.getClass)
 
-  // Name of this participant,
-  val participantId: Ref.ParticipantId = config.participantId
-
-  override def acquire()(implicit context: ResourceContext): Resource[ApiServer] = {
-    val packageStore = loadDamlPackages()
-    preloadPackages(packageStore)
+  def apply(
+      ledgerId: LedgerId,
+      config: ApiServerConfig,
+      commandConfig: CommandConfiguration,
+      partyConfig: PartyConfiguration,
+      submissionConfig: SubmissionConfiguration,
+      optWriteService: Option[state.WriteService],
+      authService: AuthService,
+      healthChecks: HealthChecks,
+      metrics: Metrics,
+      timeServiceBackend: Option[TimeServiceBackend] = None,
+      otherServices: immutable.Seq[BindableService] = immutable.Seq.empty,
+      otherInterceptors: List[ServerInterceptor] = List.empty,
+      engine: Engine,
+      servicesExecutionContext: ExecutionContextExecutor,
+      lfValueTranslationCache: LfValueTranslationCache.Cache,
+      checkOverloaded: TelemetryContext => Option[state.SubmissionResult] =
+        _ => None, // Used for Canton rate-limiting
+  )(implicit
+      actorSystem: ActorSystem,
+      materializer: Materializer,
+      loggingContext: LoggingContext,
+  ): ResourceOwner[ApiServer] = {
+    for {
+      indexService <- indexService(
+        ledgerId = ledgerId,
+        config = config,
+        metrics = metrics,
+        engine = engine,
+        servicesExecutionContext = servicesExecutionContext,
+        lfValueTranslationCache = lfValueTranslationCache,
+      )
+      apiServer <- from(
+        indexService = indexService,
+        ledgerId = ledgerId,
+        config = config,
+        commandConfig = commandConfig,
+        partyConfig = partyConfig,
+        submissionConfig = submissionConfig,
+        optWriteService = optWriteService,
+        authService = authService,
+        healthChecks = healthChecks,
+        metrics = metrics,
+        timeServiceBackend = timeServiceBackend,
+        otherServices = otherServices,
+        otherInterceptors = otherInterceptors,
+        engine = engine,
+        servicesExecutionContext = servicesExecutionContext,
+        checkOverloaded = checkOverloaded,
+      )
+    } yield apiServer
+  }
 
+  def indexService(
```

Comment:
Does this need to be public?

If it does, can we factor it into its own file?

Reply:
I think this being public is the whole point of this refactoring (see [comment](https://github.com/digital-asset/daml/pull/11842#issuecomment-977593551)).

Reply:
Then why not put it in its own object?